src/mpi/init/netloc_util.c

```diff
@@ -280,6 +287,7 @@ static int get_tree_attributes(netloc_topology_t topology,
     }
 
   fn_exit:
+    MPIR_CHKPMEM_COMMIT();
```

Comment:
We should add `MPIR_CHKPMEM_REAP()` after the `fn_fail` label to complete the error handling logic.

Reply:
@raffenet This PR is rebased on my other PRs, I'll rename my PRs so that they can be reviewed and merged in that order.

Reply:
I will update this in all PRs and tag you once it is done, thanks.